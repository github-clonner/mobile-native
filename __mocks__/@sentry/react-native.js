export const captureMessage = jest.fn();
export default {};
