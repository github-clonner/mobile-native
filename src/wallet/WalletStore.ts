//@ts-nocheck
import { observable, computed, action } from 'mobx';

import moment from 'moment';
import walletService from './WalletService';
import abbrev from '../common/helpers/abbrev';
import token from '../common/helpers/token';
import number from '../common/helpers/number';
import TokensStore from './tokens/TokensStore';
import storageService from '../common/services/storage.service';
import web3Service from '../blockchain/services/Web3Service';

/**
 * Wallet store
 */
class WalletStore {
  @observable balance = -1;
  @observable addresses = [];
  @observable overview = {};

  @observable onboardingShown = false;

  ledger = new TokensStore();

  refreshing = false;
  loaded = false;

  @action
  clockTick() {
    this.overview.nextPayout--;
  }

  @action
  async refresh(force = false) {
    if ((this.refreshing || this.loaded) && !force) {
      return;
    }

    this.refreshing = true;

    const { balance, addresses } = await walletService.getBalances();

    if (addresses && addresses.length > 0) {
      addresses.forEach(async (address) => {
        if (
          address.label.toLowerCase() != 'offchain' &&
          address.address !== ''
        ) {
          address.ethBalance = await web3Service.getBalance(address.address);
        }
      });
    }

    const overview = await walletService.getContributionsOverview();

    // next payout clock
    this.interval && clearInterval(this.interval);
    this.interval = setInterval(() => this.clockTick(), 1000);

    this.overview = overview;
    this.balance = balance;
    this.addresses = addresses;

    this.refreshing = false;
    this.loaded = true;
  }

  @computed get formattedBalance() {
    return this.balance > -1 ? number(token(this.balance, 18), 3) : 'â€¦';
  }

  /**
   * Join to wallet tokens
   * @param {string} number
   * @param {boolean} retry
   */
  join(number, retry) {
    return walletService.join(number, retry);
  }

  /**
   * Confirm join
   * @param {string} number
   * @param {string} code
   * @param {string} secret
   */
  confirm(number, code, secret) {
    return walletService.confirm(number, code, secret);
  }

  @action reset() {
    this.balance = -1;
    this.addresses = [];
    this.refreshing = false;
    this.loaded = false;
    this.overview = {
      contributionValues: {
        comments: 2,
        reminds: 4,
        votes: 1,
        subscribers: 4,
        referrals: 50,
        referrals_welcome: 50,
        checkin: 2,
        jury_duty: 25,
      },
    };
    this.onboardingShown = false;
    this.ledger = new TokensStore();
    storageService.removeItem('walletOnboardingComplete');
  }

  // TODO: Implement forced auto-refresh every X minutes
  // TODO: Implement socket and atomic increases/decreases

  // Onboarding

  async canShowOnboarding() {
    return (
      !this.isOnboardingShown &&
      !(await storageService.getItem('walletOnboardingComplete'))
    );
  }

  @action setOnboardingShown(value) {
    this.isOnboardingShown = !!value;
  }

  async setOnboardingComplete(value) {
    await storageService.setItem('walletOnboardingComplete', !!value);
  }
}

export default WalletStore;
