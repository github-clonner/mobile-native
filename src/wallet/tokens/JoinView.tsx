//@ts-nocheck
import React, { Component } from 'react';

import {
  Text,
  TextInput,
  View,
  TouchableHighlight,
  StyleSheet,
} from 'react-native';

import { observer, inject } from 'mobx-react';

import PhoneInput from 'react-native-phone-input';

import { CommonStyle } from '../../styles/Common';
import { ComponentsStyle } from '../../styles/Components';
import Button from '../../common/components/Button';
import i18n from '../../common/services/i18n.service';

/**
 * Token & rewards join view
 */
@inject('wallet', 'user')
export default class JoinView extends Component {
  state = {
    confirming: false,
    confirmation: '',
    invalidNumber: false,
    number: '',
    secret: '',
    error: '',
  };

  /**
   * Reset state
   */
  reset() {
    this.setState({
      confirming: false,
      confirmation: '',
      invalidNumber: false,
      number: '',
      secret: '',
      error: '',
    });
  }

  /**
   * Join
   */
  join = () => {
    const valid = this.refs.phone.isValidNumber();

    if (!valid) {
      this.setState({ invalidNumber: true });
      return;
    }

    const number = this.refs.phone.getValue();

    this.props.wallet
      .join(number)
      .then(({ secret }) => {
        this.setState({
          secret,
          number,
          confirming: true,
        });
      })
      .catch((e) => {
        this.setState({ error: e.message });
      });
  };

  /**
   * Confirm
   */
  confirm = () => {
    this.props.wallet
      .confirm(this.state.number, this.state.confirmation, this.state.secret)
      .then((response) => {
        this.props.user.setRewards(true);
      })
      .catch((e) => {
        this.setState({ error: e.message });
      });
  };

  /**
   * Render
   */
  render() {
    const body = this.getBody();

    const error = this.state.error ? (
      <Text style={[CommonStyle.fontL, CommonStyle.colorDanger]}>
        {this.state.error}
      </Text>
    ) : null;

    return (
      <View>
        <Text
          style={[
            CommonStyle.fontS,
            { marginTop: 8, color: '#444', padding: 4 },
          ]}>
          {i18n.t('wallet.join.rewardsDescription')}
        </Text>

        {error}
        {body}

        <Text
          style={[
            CommonStyle.fontS,
            { marginTop: 20, color: '#444', padding: 4 },
          ]}>
          {i18n.t('wallet.join.note')}
        </Text>
      </View>
    );
  }

  /**
   * Change confirmation value
   */
  changeConfirmation = (val) => {
    this.setState({ confirmation: val });
  };

  /**
   * Cancel
   */
  cancel = () => {
    this.reset();
  };

  /**
   * Get body
   */
  getBody() {
    if (this.state.confirming) {
      return (
        <View>
          <View style={[CommonStyle.marginTop3x, CommonStyle.marginBottom3x]}>
            <TextInput
              ref="input"
              onChangeText={this.changeConfirmation}
              style={[ComponentsStyle.input, { fontSize: 16 }]}
              underlineColorAndroid="transparent"
              value={this.state.confirmation}
              placeholder={i18n.t('wallet.join.pleaseEnterCode')}
              keyboardType="numeric"
            />
          </View>
          <View
            style={{
              flexDirection: 'row',
              alignItems: 'stretch',
              marginTop: 8,
            }}>
            <View
              style={{
                flex: 1,
                flexDirection: 'column',
                alignItems: 'center',
                justifyContent: 'center',
                paddingRight: 32,
                paddingLeft: 2,
              }}>
              <Text style={[CommonStyle.fontXS, { color: '#444' }]}>
                {i18n.t('wallet.join.pleaseEnterCodeDescription')}
              </Text>
            </View>

            <View style={[CommonStyle.rowJustifyStart]}>
              <TouchableHighlight
                underlayColor="transparent"
                onPress={this.cancel}
                style={[
                  ComponentsStyle.button,
                  { backgroundColor: 'transparent', marginRight: 4 },
                ]}>
                <Text
                  style={[CommonStyle.paddingLeft, CommonStyle.paddingRight]}>
                  {i18n.t('cancel')}
                </Text>
              </TouchableHighlight>
              <TouchableHighlight
                underlayColor="transparent"
                onPress={this.confirm}
                style={[
                  ComponentsStyle.button,
                  ComponentsStyle.buttonAction,
                  { backgroundColor: 'transparent' },
                ]}>
                <Text
                  style={[
                    CommonStyle.paddingLeft,
                    CommonStyle.paddingRight,
                    CommonStyle.colorPrimary,
                  ]}>
                  {i18n.t('confirm')}
                </Text>
              </TouchableHighlight>
            </View>
          </View>
        </View>
      );
    } else {
      return (
        <View>
          <View style={[CommonStyle.marginTop3x, CommonStyle.marginBottom3x]}>
            <View style={ComponentsStyle.input}>
              <PhoneInput
                ref="phone"
                textStyle={{ letterSpacing: 18, fontSize: 16 }}
              />
            </View>
          </View>

          <View
            style={{
              flexDirection: 'row',
              alignItems: 'stretch',
              marginTop: 8,
            }}>
            <View
              style={{
                flex: 1,
                flexDirection: 'column',
                alignItems: 'center',
                justifyContent: 'center',
                paddingRight: 56,
                paddingLeft: 2,
              }}>
              {this.state.invalidNumber && (
                <Text style={[CommonStyle.fontS, CommonStyle.colorDanger]}>
                  {i18n.t('wallet.join.numberInvalid')}
                </Text>
              )}
              <Text style={[CommonStyle.fontXS, { color: '#444' }]}>
                {i18n.t('wallet.join.pleaseEnterPhone')}
              </Text>
            </View>
            <View>
              <TouchableHighlight
                underlayColor="transparent"
                onPress={this.join}
                style={[
                  ComponentsStyle.button,
                  ComponentsStyle.buttonAction,
                  { backgroundColor: 'transparent' },
                ]}>
                <Text
                  style={[
                    CommonStyle.paddingLeft,
                    CommonStyle.paddingRight,
                    CommonStyle.colorPrimary,
                  ]}>
                  {i18n.t('join')}
                </Text>
              </TouchableHighlight>
            </View>
          </View>
        </View>
      );
    }
  }
}

const styles = StyleSheet.create({});
