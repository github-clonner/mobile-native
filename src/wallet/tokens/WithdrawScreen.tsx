//@ts-nocheck
import React, { Component } from 'react';

import {
  View,
  Text,
  TextInput,
  ActivityIndicator,
  FlatList,
  StyleSheet,
} from 'react-native';

import { observer, inject } from 'mobx-react';

import JoinView from './JoinView';

import WithdrawService from './WithdrawService';

import TransparentButton from '../../common/components/TransparentButton';

import Colors from '../../styles/Colors';
import number from '../../common/helpers/number';
import formatDate from '../../common/helpers/date';
import token from '../../common/helpers/token';
import readableError from '../../common/helpers/readable-error';
import i18n from '../../common/services/i18n.service';
import { CommonStyle } from '../../styles/Common';

@inject('user', 'withdraw')
@observer
export default class WithdrawScreen extends Component {
  state = {
    amount: '0',
    inProgress: false,
    error: '',
    hasWithdrawnToday: false,
    balance: 0,
    available: 0,
    withholding: 0,
  };

  static navigationOptions = {
    title: 'Withdraw',
  };

  async componentWillMount() {
    return await Promise.all([this.load(), this.loadLedger()]);
  }

  async load() {
    try {
      this.setState({ inProgress: true });

      await Promise.all([this.checkPreviousWithdrawals(), this.getBalance()]);
    } catch (e) {
      const error =
        (e && e.message) || i18n.t('wallet.withdraw.errorReadingStatus');
      this.setState({ error });
    } finally {
      this.setState({ inProgress: false });
    }
  }

  async loadLedger() {
    await this.props.withdraw.load();
  }

  async checkPreviousWithdrawals() {
    const hasWithdrawnToday = !(await WithdrawService.canWithdraw());

    this.setState({ hasWithdrawnToday });

    if (hasWithdrawnToday) {
      throw new Error(i18n.t('wallet.withdraw.errorOnlyOnceDay'));
    }
  }

  async getBalance() {
    const { balance, available } = await WithdrawService.getBalance();

    let withholding = 0;

    if (balance > available) {
      withholding = balance - available;
    }

    this.setState({ balance, available, withholding });
    this.setAmount(available);
  }

  canWithdraw() {
    return (
      !this.state.hasWithdrawnToday &&
      !this.state.inProgress &&
      parseFloat(this.state.amount) > 0 &&
      parseFloat(this.state.amount) <= this.state.available
    );
  }

  async withdraw() {
    this.setState({ inProgress: true, error: '' });

    try {
      await this.props.withdraw.withdraw(
        this.props.user.me.guid,
        parseFloat(this.state.amount),
      );
      this.setState({ amount: '0' });
    } catch (e) {
      if (!e || e.message !== 'E_CANCELLED') {
        const error =
          (e && e.message) || i18n.t('wallet.withdraw.errorWithdrawing');
        this.setState({ error: readableError(error) });
      }
    } finally {
      this.setState({ inProgress: false });
    }
  }

  withdrawAction = () => this.withdraw();

  setAmount = (amount) => {
    if (typeof amount === 'string') {
      amount = amount.replace(/,/g, '');
    }

    if (
      typeof amount === 'string' &&
      amount.substring(amount.length - 1) === '.'
    ) {
      // Nothing, since we're inputting numbers, let's keep the dot
    } else if (
      typeof amount === 'string' &&
      !isNaN(+amount - parseFloat(amount))
    ) {
      amount = `${+amount}`;
    }

    this.setState({ amount });
  };

  getAmount() {
    if (!this.state.amount && this.state.amount !== 0) {
      return '';
    }

    let withDot = false;
    if (
      typeof this.state.amount === 'string' &&
      this.state.amount.substring(this.state.amount.length - 1) === '.'
    ) {
      withDot = true;
    }

    try {
      return `${number(this.state.amount, 0, 4)}` + (withDot ? '.' : '');
    } catch (e) {
      return `${this.state.amount}`;
    }
  }

  getFormPartial = () => {
    let withdrawButonContent = 'WITHDRAW';

    if (this.state.inProgress) {
      withdrawButonContent = (
        <ActivityIndicator size="small" color={Colors.primary} />
      );
    }

    return (
      <View style={style.formWrapperView}>
        <Text style={style.legendText}>
          {i18n.to(
            'wallet.withdraw.youCanRequest1',
            {
              amount: this.getAmount(),
            },
            {
              onchain: <Text style={CommonStyle.bold}>OnChain</Text>,
            },
          )}
          {'\n'}
          <Text style={{ fontSize: 11 }}>
            {i18n.to('wallet.withdraw.youCanRequest2', null, {
              approval: (
                <Text style={CommonStyle.bold}>
                  {i18n.t('wallet.withdraw.youCanRequest3')}
                </Text>
              ),
            })}
          </Text>
          {!!this.state.withholding
            ? i18n.t('wallet.withdraw.holdingMessage', {
                amount: number(this.state.withholding, 0, 4),
              })
            : ''}
        </Text>

        <View style={style.formView}>
          <TextInput
            editable={!this.state.inProgress}
            style={style.amountTextInput}
            placeholder={i18n.t('wallet.withdraw.amount')}
            keyboardType="numeric"
            returnKeyType="go"
            onChangeText={this.setAmount}
            value={this.getAmount()}
          />

          <TransparentButton
            disabled={!this.canWithdraw()}
            style={style.withdrawButton}
            color={Colors.primary}
            disabledColor={Colors.greyed}
            title={withdrawButonContent}
            onPress={this.withdrawAction}
          />
        </View>

        {!!this.state.error && (
          <Text style={style.errorText}>{this.state.error}</Text>
        )}
      </View>
    );
  };

  getLedgerItemPartial = ({ item }) => {
    return (
      <View style={style.ledgerItem}>
        <View style={style.ledgerItemMeta}>
          <Text style={style.ledgerItemDate}>
            {formatDate(item.timestamp, 'date')}
          </Text>
          <Text style={style.ledgerItemStatus}>
            {item.completed
              ? i18n.t('completed').toUpperCase()
              : i18n.t('pending').toUpperCase()}
          </Text>
        </View>

        <Text style={style.ledgerItemAmount}>
          {number(token(item.amount), 0, 3)}
        </Text>
      </View>
    );
  };

  ledgerKeyExtractor = (item) => `${item.timestamp}_${item.tx}`;

  render() {
    const ledger = this.props.withdraw.ledger,
      formPartial = this.getFormPartial();

    if (!this.props.user.me.rewards) {
      return (
        <View style={style.view}>
          <JoinView />
        </View>
      );
    }

    return (
      <FlatList
        style={style.view}
        data={ledger}
        renderItem={this.getLedgerItemPartial}
        keyExtractor={this.ledgerKeyExtractor}
        ListHeaderComponent={formPartial}
      />
    );
  }
}

const style = StyleSheet.create({
  view: {
    flex: 1,
    padding: 10,
    backgroundColor: '#fff',
  },
  formWrapperView: {
    paddingBottom: 30,
  },
  legendText: {
    color: Colors.darkGreyed,
    marginBottom: 20,
    letterSpacing: 0.35,
  },
  formView: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  amountTextInput: {
    flexGrow: 1,
    marginRight: 10,
    padding: 8,
    fontSize: 18,
    fontWeight: '700',
    textAlign: 'right',
    backgroundColor: '#eee',
    borderRadius: 3,
  },
  withdrawButton: {
    minWidth: 100,
  },
  errorText: {
    marginTop: 20,
    color: '#c00',
    fontSize: 14,
    textAlign: 'center',
  },
  ledgerItem: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingBottom: 20,
  },
  ledgerItemMeta: {
    flexGrow: 1,
  },
  ledgerItemDate: {
    fontWeight: '500',
    letterSpacing: 0.5,
    marginBottom: 3,
  },
  ledgerItemStatus: {
    fontSize: 11,
    letterSpacing: 0.35,
    color: Colors.darkGreyed,
  },
  ledgerItemAmount: {
    fontSize: 24,
    fontWeight: '700',
    color: 'green',
  },
});
