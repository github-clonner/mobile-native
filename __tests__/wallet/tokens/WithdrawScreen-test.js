import 'react-native';
import React from 'react';

import { configure } from 'mobx';
import WithdrawScreen from '../../../src/wallet/tokens/WithdrawScreen';
import UserStore from '../../../src/auth/UserStore';
import WalletStore from '../../../src/wallet/WalletStore';
import { shallow } from 'enzyme';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import componentOrWrappedComponent from '../../../tests-helpers/component-or-wrapped-component';
import { FlatList } from 'react-native';
import JoinView from '../../../src/wallet/tokens/JoinView';

jest.mock('../../../src/auth/UserStore');
jest.mock('../../../src/wallet/WalletStore');


configure({enforceActions: 'never'});

describe('WithdrawScreen', () => {
  let user, withdraw, screen;

  beforeEach(() => {
    user = new UserStore();
    withdraw = new WalletStore();

    user.me.rewards = true;

    screen = shallow(
      <WithdrawScreen.wrappedComponent user={user} withdraw={withdraw} />
    );
  });

  it('renders correctly', () => {
    const renderedScreen = renderer.create(
      <WithdrawScreen.wrappedComponent user={user} withdraw={withdraw} />
    );

    expect(renderedScreen.toJSON()).toMatchSnapshot();
  });

  it('should have a flatlist', async () => {
    screen.update();

    expect(screen.find(FlatList)).toHaveLength(1);
  });

  it('should show join component', async () => {
    user.me.rewards = false;

    // we call willmount again because user.me.rewards was updated
    await screen.instance().componentWillMount();
    screen.update();

    expect(screen.find((JoinView)))
      .toHaveLength(1);
  })
});
