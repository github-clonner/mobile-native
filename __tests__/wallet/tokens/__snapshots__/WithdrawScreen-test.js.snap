// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithdrawScreen renders correctly 1`] = `
<RCTScrollView
  ListHeaderComponent={
    <View
      style={
        Object {
          "paddingBottom": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#777777",
            "letterSpacing": 0.35,
            "marginBottom": 20,
          }
        }
      >
        You can request to withdraw up to 0 tokens from your rewards to your 
        <Text
          style={
            Object {
              "fontWeight": "700",
            }
          }
        >
          OnChain
        </Text>
         wallet.
        

        <Text
          style={
            Object {
              "fontSize": 11,
            }
          }
        >
          Note: a small amount of ETH will be charged to cover the transaction fee. Withdrawals 
          <Text
            style={
              Object {
                "fontWeight": "700",
              }
            }
          >
            go through an approval process
          </Text>
           and may take up to 72 hours to complete
        </Text>
        
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={false}
          keyboardType="numeric"
          onChangeText={[Function]}
          placeholder="Amount"
          rejectResponderTermination={true}
          returnKeyType="go"
          style={
            Object {
              "backgroundColor": "#eee",
              "borderRadius": 3,
              "flexGrow": 1,
              "fontSize": 18,
              "fontWeight": "700",
              "marginRight": 10,
              "padding": 8,
              "textAlign": "right",
            }
          }
          underlineColorAndroid="transparent"
          value="0"
        />
        <TransparentButton
          color="#4690D6"
          disabled={true}
          disabledColor="#cccccc"
          onPress={[Function]}
          style={
            Object {
              "minWidth": 100,
            }
          }
          title={
            <ActivityIndicator
              animating={true}
              color="#4690D6"
              hidesWhenStopped={true}
              size="small"
            />
          }
        />
      </View>
    </View>
  }
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 10,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "paddingBottom": 30,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#777777",
              "letterSpacing": 0.35,
              "marginBottom": 20,
            }
          }
        >
          You can request to withdraw up to 0 tokens from your rewards to your 
          <Text
            style={
              Object {
                "fontWeight": "700",
              }
            }
          >
            OnChain
          </Text>
           wallet.
          

          <Text
            style={
              Object {
                "fontSize": 11,
              }
            }
          >
            Note: a small amount of ETH will be charged to cover the transaction fee. Withdrawals 
            <Text
              style={
                Object {
                  "fontWeight": "700",
                }
              }
            >
              go through an approval process
            </Text>
             and may take up to 72 hours to complete
          </Text>
          
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={false}
            keyboardType="numeric"
            onChangeText={[Function]}
            placeholder="Amount"
            rejectResponderTermination={true}
            returnKeyType="go"
            style={
              Object {
                "backgroundColor": "#eee",
                "borderRadius": 3,
                "flexGrow": 1,
                "fontSize": 18,
                "fontWeight": "700",
                "marginRight": 10,
                "padding": 8,
                "textAlign": "right",
              }
            }
            underlineColorAndroid="transparent"
            value="0"
          />
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "borderColor": "#ececec",
                  "borderRadius": 30,
                  "borderWidth": 1,
                  "padding": 8,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "minWidth": 100,
                },
                Object {
                  "borderColor": "#4690D6",
                },
                Object {
                  "borderColor": "#cccccc",
                },
              ]
            }
          >
            <ActivityIndicator
              animating={true}
              color="#4690D6"
              hidesWhenStopped={true}
              size="small"
              style={null}
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
